### ISSUE TITLE
ImportError: cannot import name 'Bidirectional' from 'tensorflow.python.keras.layers' (tf-nightly)

### ISSUE THREAD

**Issue by Mhbuur** [tensorflow/tensorflow#54180]
I get the following build error on Win10, using tf-nightly 2.9.0.dev20220127 with Python 3.10.2, no GPU:

ImportError: cannot import name 'Bidirectional' from 'tensorflow.python.keras.layers' (C:\Python310\lib\site-packages\tensorflow\python\keras\layers\__init__.py)

I'm using VS Code and as such the import resolves just fine. I had to change the import line from
`tensorflow.keras.layers` to `tensorflow.python.keras.layers` though.

Am I doing something wrong here or is it an issue with tf-nightly and win10/python 3.10?

**Comment by sushreebarsa** [comment]
@Mhbuur 
In order to expedite the trouble-shooting process here,Could you please fill the issue [template](https://github.com/tensorflow/tensorflow/issues/new/choose),
Thanks!

**Comment by Mhbuur** [comment]
@sushreebarsa better? At this point I am not really sure if this should be reported as a bug or a build/install issue or  general support. Enlighten me please :-)

**Comment by sushreebarsa** [comment]
@Mhbuur Could you please try with the latest TF v2.7.0 and refer to [build from source](https://www.tensorflow.org/install/source_windows) , [tested build configurations](https://www.tensorflow.org/install/source_windows#tested_build_configurations) ? Please find the colab [gist](https://colab.research.google.com/gist/sushreebarsa/e59f95fa9601131fbe9eed33ff2811bd/untitled551.ipynb#scrollTo=F-A5eL_ntWKA) for reference and let us know if it helps?Thanks!

**Comment by Mhbuur** [comment]
@sushreebarsa Python 3.10 is not supported in TF 2.7.0, which is why I'm trying out 2.9 / tf-nightly

**Comment by sushreebarsa** [comment]
@Mhbuur Please see the discussion on [TF Forum ](https://discuss.tensorflow.org/t/support-python-3-10/124?u=mihaimaruseac).Thanks!

**Comment by Mhbuur** [comment]
@sushreebarsa thanks. I am aware of that discussion - it got me started on tf-nightly ;-) Should I report there instead?

**Comment by sushreebarsa** [comment]
@Mhbuur I tried to replicate the issue on colab using tf-nightly(2.9.0.dev20220131) but didn't face the `ImportError` as reported.Please find the gist [here](https://colab.research.google.com/gist/sushreebarsa/e59f95fa9601131fbe9eed33ff2811bd/untitled551.ipynb#scrollTo=6jNR5acnpAjJ) and confirm the same ?
 `tensorflow.python` libraries are used for internal developing purpose so it is not recommended to use .  We recommend you to use `from tensorflow.keras.layers import LSTM, Dense, Dropout, Bidirectional `
Thanks!

**Comment by Mhbuur** [comment]
@sushreebarsa how do you get hold of tf-nightly version 2.9.0.dev20220131? 
When update or install using --no-cache param (after uninstalling) I keep getting tf_nightly 2.9.0.dev20220129
![image](https://user-images.githubusercontent.com/419020/151825547-d5a2e1e8-b350-4b0b-bef6-eed721015c57.png)

Regarding the import path: the only way I could get VS Code to resolve the import was to use the `tensorflow.python.keras.layers` path. If I use `tensorflow.keras.layers` it will not resolve and I get new runtime errors. Do you think this is the cause for the first ImportError problem regarding 'Bidirectional'?

**Comment by Mhbuur** [comment]
@sushreebarsa ok, there is some progress - after updating I can actually run the script (without the tensorflow.python path) - it still doesn't resolve in the IDE, but the script now runs without the bidirectional error. So either the update/reinstall fixed it or the import path change, I can't tell for sure.

The VS Code import path resolve issue is discussed many other places, so I'll close this issue for now.

**Comment by macoskey** [comment]
@Mhbuur could you please post a link to a solution for "the VS Code import path resolve issue" if you found one?

**Comment by macoskey** [comment]
SOLUTION: See [here](https://github.com/microsoft/pylance-release/issues/1066#issuecomment-1022431087). You may want to add other things like `layer`, `models`, `activations`, `regularizers`, `constraints`, etc.

This is so hacky though... Feels like the TensorFlow team should take a look at this.

**Comment by Mhbuur** [comment]
@macoskey Yes, that was one of the suggestions I looked at. I didn't try it out as I found it a bit too hacky. But thanks for confirming that it works as a temporary measure.
